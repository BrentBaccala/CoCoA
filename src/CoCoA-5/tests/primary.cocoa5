PrintLn "-- TEST 01 :";

    Use QQ[x,y,z];
    I := ideal(x^3-y*z, y^2-x*z, z^2-x^2*y);
    TgCone(I);
    InitialIdeal(I, [y,z]);
    Q := ideal(y, z);
    PS := PrimaryPoincare(I, Q);
    $hp.PSerToHilbert(PS); -- the Hilbert Function associated to PS

PrintLn "-- TEST 02 :";

    Use ZZ/(32003)[x,y,z,w];
    I := ideal(x^5 - y*z, y^4 - x*z^2, x*y^3 - z*w, x^2*z - y*w,
    y^2*z^2 - w^3, y^3*z - x^2*w^2, x^3*w - z^2, x*y*w^2 - z^3,
    x^3*y^2 - w^2, x*z^4 - y^2*w^3, y*z^5 - x*w^5, y^3*w^5 - z^7,
    x^2*w^7 - z^8, z^9 - y*w^8);
    TgCone(I);
    Q  := ideal(x, y, z);
    PS := PrimaryPoincare(I, Q); PS;
    $hp.PSerToHilbert(PS);

PrintLn "-- TEST 03 :";

    Use P ::= ZZ/(32003)[x,y,z];
    I  := ideal(P,[]);
    Q  := ideal(x, y, z^2);
    PS := PrimaryPoincare(I, Q); PS;
    $hp.PSerToHilbert(PS);
    HV := $hp.PSerHVector(PS); -- the H-vector associated to PS

PrintLn "-- TEST 04 :";

    Use ZZ/(32003)[x,y,z,w];
    I  := ideal(-y*z + x*w, z^3 - y*w^2, -x*z^2 + y^2*w, -y^3 + x^2*z);
    Q  := ideal(x, y, z^2, w^3);
    PS := PrimaryPoincare(I, Q); PS;
    $hp.PSerToHilbert(PS);
    HV := $hp.PSerHVector(PS);
    $primary.E(0, HV);
    [ $primary.E(J,HV) | J In 0..($hp.PSerDim(PS)-2) ];
    [ $primary.E(J,HV) | J In 0..(len(HV)-1) ];

PrintLn "-- TEST 05 :";

    Use ZZ/(32003)[x,y,z,w];
    I  := ideal(x^3-y^7, x^2*y - x*w^3-z^6);
    Q  := ideal(x, y, z, w);
    PS := PrimaryPoincare(I, Q);
    $hp.PSerToHilbert(PS);
    HV := $hp.PSerHVector(PS);
    [ $primary.E(J,HV) | J In 0..($hp.PSerDim(PS)-2) ];
    [ $primary.E(J,HV) | J In 0..(len(HV)-1) ];

PrintLn "-- TEST 06 :";

    Use ZZ/(32003)[x,y,z];
    I  := ideal(z^3);
    Q  := ideal(x^2, y^2, x*z, y*z);
    PS := PrimaryPoincare(I, Q); PS;
    $hp.PSerToHilbert(PS);
    HV := $hp.PSerHVector(PS);
    [ $primary.E(J,HV) | J In 0..($hp.PSerDim(PS)-2) ];
    [ $primary.E(J,HV) | J In 0..(len(HV)-1) ];
