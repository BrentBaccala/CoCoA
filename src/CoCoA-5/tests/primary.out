
# -- TEST 01 :

# 
# 
# 
# ideal(z^2, -y*z, y^2 -x*z)

# ideal(-x*z, -x^2*y, x^3, -z^4, x*y^3 -z^3)

# 
# 
# H(0) = 3
H(1) = 3
H(t) = 4   for t >= 2

# 
# -- TEST 02 :

# 
# 
# 
... ; # 
... ; # 
... ; # 
# ideal(w^2, z*w, y*w, z^2, y*z, y^4)

# 
# (2 + 2*t + t^2 + t^3) / (1-t)

# H(0) = 2
H(1) = 4
H(2) = 5
H(t) = 6   for t >= 3

# 
# -- TEST 03 :

# 
# 
# 
# 
# (2) / (1-t)^3

# H(t) = t^2 +3*t +2   for t >= 0

# 
# 
# -- TEST 04 :

# 
# 
# 
# 
# (6 + 4*t) / (1-t)^2

# H(t) = 10*t +6   for t >= 0

# 
# 10

# [10]

# [10, 4]

# 
# -- TEST 05 :

# 
# 
# 
# 
# 
# H(0) = 1
H(1) = 4
H(2) = 10
H(3) = 18
H(4) = 28
H(5) = 39
H(6) = 52
H(7) = 66
H(8) = 80
H(9) = 95
H(10) = 110
H(11) = 126
H(12) = 141
H(13) = 156
H(t) = 16*t -53   for t >= 14

# 
# [16]

# [16, 69, 212, 579, 1398, 2833, 4636, 6010, 6114, 4841, 2948, 1354, 454, 105, 15, 1]

# 
# -- TEST 06 :

# 
# 
# 
# 
# (6 + 3*t + 4*t^2 - t^3) / (1-t)^2

# H(0) = 6
H(1) = 15
H(t) = 12*t +4   for t >= 2

# 
# [12]

# [12, 8, 1, -1]

# 